File: Cano-Rodriguez-Claudio/Procesos/Java/pipe.java
import java.io.*;

public class PipeMinimal {
    public static void main(String[] args) throws IOException {
        PipedInputStream in = new PipedInputStream();
        PipedOutputStream out = new PipedOutputStream(in);

        Thread child = new Thread(() -> {
            try {
                byte[] buffer = new byte[5];
                in.read(buffer);
                System.out.println("H: " + new String(buffer));
            } catch (IOException e) {
                e.printStackTrace();
            }
        });

        child.start();

        // Parent process (main thread)
        String msg = "abcde";
        out.write(msg.getBytes());
        System.out.println("P: " + msg);
    }
}

File: Cano-Rodriguez-Claudio/Procesos/C/pipe.c
#include <stdio.h>
#include <unistd.h>
#include <string.h>

int main() {
    int fd[2];
    char buf[] = "abcde", rbuf[6]; // Buffer for writing and reading
    pipe(fd);

    if (fork() == 0) { // Child process
        close(fd[1]); // Close write end
        read(fd[0], rbuf, 5);
        rbuf[5] = '\0';
        printf("H: %s\n", rbuf);
    } else { // Parent process
        close(fd[0]); // Close read end
        write(fd[1], buf, 5);
        printf("P: %s\n", buf);
    }
    return 0;
}
